### ISSUE TITLE
@conventionDeclarationReference

### ISSUE THREAD

**Issue by SJagodzinski** [SAA-SDT/eac-cpf-schema#174]
## Convention declaration reference

Add new optional attribute `@conventionDeclarationReference` with data type IDREFS in all non-empty elements alongside the language attributes `@lanuageOfElement` and `@scriptOfElement `.

With `@conventionDeclarationReference` a direct link to a conventation declaration element within cpf/control/conventionDeclaration can be provided. The attribute can be used to link to a convention or rule that prescribe a method for converting one script into another script (transliteration). It also can be used to link to a national, international or other rule that govern the construction of EAC-CPF name (formerly used in dropped elements `<alternativeForm>`, `<authorizedForm>`, and `preferredForm`).

## Creator of issue
1. Silke Jagodzinski
2. TS-EAS: EAC-CPF subgroup
3. silkejagodzinski@gmail.com

## Related issues / documents
Topic: Assertion Description in EAS
Assertion Description (or Assertion Control) #43
[Topic_names_20200130.pdf](https://github.com/SAA-SDT/TS-EAS-subteam-notes/blob/master/eaccpf-subteam/working-documents/topics/names/Topic_names_20200130.pdf)
Names control #26

## EAD3 Reconciliation
new EAS attribute

## Context
new EAS attribute

## Solution documentation: agreed solution for TL and guidelines
Rephrasing _Summary_, _Description and Usage_ and _Attribute usage_ needed

**Data type**: IDREF

**May occur within**: `<abstract>`, `<address>`, `<addressLine>`, `<agencyCode>`, `<agencyName>`, `<agent>`, `<alternativeSet>`, `<biogHist>`, `<chronItem>`, `<chronItemSet>`, `<chronList>`, `<citedRange>`, `<componentEntry>`, `<contact>`, `<contactLine>`, `<conventionDeclaration>`, `<date>`, `<dateRange>`, `<dateSet>`, `<description>`, `<descriptiveNote>`, `<event>`, `<eventDateTime>`, `<eventDescription>`, `<existDates>`, `<fromDate>`, `<function>`, `<functions>`, `<generalContext>`, `<geographicCoordinates>`, `<head>`, `<identityId>`, `<item>`, `<language>`, `<languageDeclaration>`, `<languageUsed>`, `<languagesUsed>`, `<legalStatus>`, `<legalStatuses>`, `<list>`, `<localControl>`, `<localDescription>`, `<localDescriptions>`, `<localTypeDeclaration>`, `<maintenanceAgency>`, `<maintenanceEvent>`, `<maintenanceHistory>`, `<mandate>`, `<mandates>`, `<nameEntry>`, `<nameEntrySet>`, `<occupation>`, `<occupations>`, `<otherAgencyCode>`, `<otherEntityType>`, `<otherEntityTypes>`, `<otherRecordId>`, `<p>`, `<part>`, `<place>`, `<placeName>`, `<placeRole>`, `<places>`, `<recordId>`, `<reference>`, `<relation>`, `<relationType>`, `<representation>`, `<rightsDeclaration>`, `<setComponent>`, `<shortCode>`, `<source>`, `<sources>`, `<span>`, `<structureOrGenealogy>`, `<targetEntity>`, `<targetRole>`, `<term>`, `<toDate>`, `<useDates>`, `<writingSystem>`

- Example needed
- Topic for Best Practise Guide

## Example encoding

**Comment by ailie-s** [comment]
Tag Library Text:

**Summary:** Use `@conventionDeclarationReference` to provide a direct link to a `<conventionDeclaration>` element within control/conventionDeclaration from any non-empty elements in the EAC-CPF instance. The attribute can be used to link to a convention or rule that prescribes a method for converting one script into another script (transliteration). It also can be used to link to a national, international or other rule that governs the construction of an EAC-CPF name.
**Data type:** IDREFS

**Comment by ailie-s** [comment]
Tag Library Text:

**Summary:** Use `@localTypeDeclarationReference` to provide a direct link to a `<localTypeDeclaration>` element within <gi>control</gi> from another element using `@localType` within the EAC-CPF instance. `@localTypeDeclarationReference` should always be used when `@localType` is used, in order to link to the local type definition.
**Data type:** IDREF

**Comment by deletedname** [comment]
Spelling error: **conventation**Declaration in the header and body of text

**Comment by SJagodzinski** [comment]
@deletedname : thanks a lot ;-)

**Comment by fordmadox** [comment]
Regarding the datatype:  @ailie-s @SJagodzinski @kerstarno,  I won't tag each issue unless there is agreement on this approach, but based on the discussion around how to convert the nameEntry / nameEntryParallel elements from EAC 1.x to 2.0 (e.g. https://github.com/SAA-SDT/eac-cpf-schema/issues/117#issuecomment-763028683), I changed the datatypes for all of the *Reference attributes and `@target` from IDREF to **IDREFS**.  Does that sound like a good approach for the call for comments / migration process?

**Comment by kerstarno** [comment]
There currently are some inconsistencies with regard to where `@conventionDeclarationReference` and its two sibling attributes, `@maintenanceEventReference` (#173)  and `@sourceReference` (#172), can be used:

- The solution documentation above lists 81 elements, which are the same as for `@languageOfElement`(#151) and `@scriptOfElement`(#152). Similarly, the TL draft above says, `@conventionDeclarationReference` can be used "from any non-empty elements in the EAC-CPF instance".
- In the draft schema (RNG and XSD), however, there are currently only 63 elements that allow for the usage of `@conventionDeclarationReference` and its two sibling attributes.
  - This adds `<cpfDescription>`, `<identity>`, and `<relations>`, which are not listed here, but do have the `@...Reference` attributes as well as the language attributes at the moment.
  - And it excludes the elements that only appear in `<control>` (i.e. `<agencyCode>`, `<agencyName>`, `<agent>`, `<citedRange>`, `<conventionDeclaration>`, `<eventDateTime>`, `<eventDescription>`, `<languageDeclaration>`, `<localControl>`, `<localTypeDeclaration>`, `<maintenanceAgency>`, `<maintenanceEvent>`, `<maintenanceHistory>`, `<otherAgencyCode>`, `<otherRecordId>`, `<recordId>`, `<representation>` , `<rightsDeclaration>`, `<shortCode>`, `<source>`, `<sources>`), which is in line with the general decision on this taken during the Chicago meeting.
  - @SJagodzinski Could you please confirm where `@conventionDeclarationReference` and its two sibling attributes should be used? 

Apart from that:

- `@conventionDeclarationReference` has the data type _IDREFS_
- It is always available alongside `@maintenanceEventReference`
- It is always available alongside `@sourceReference`

The above applies to both schemas, RNG and XSD.

@ailie-s the data type has recently changed to IDREFS in order to allow for several IDs to be pointed to within one attribute.

**Comment by SJagodzinski** [comment]
> Regarding the datatype: @ailie-s @SJagodzinski @kerstarno, I won't tag each issue unless there is agreement on this approach, but based on the discussion around how to convert the nameEntry / nameEntryParallel elements from EAC 1.x to 2.0 (e.g. [#117 (comment)](https://github.com/SAA-SDT/eac-cpf-schema/issues/117#issuecomment-763028683)), I changed the datatypes for all of the *Reference attributes and `@target` from IDREF to **IDREFS**. Does that sound like a good approach for the call for comments / migration process?

@fordmadox : I agree and true, no need to comment each issues with this solution.

**Comment by SJagodzinski** [comment]
Schema version [85f90b0](https://github.com/SAA-SDT/eac-cpf-schema/commit/85f90b037a59063d7bc56a6efb4a02e2577adab9) uses reference attributes correctly, no need to change. 

Documentation will be updated.

**Comment by kerstarno** [comment]
@SJagodzinski thanks for the confirmation. 

With this, the attribute is ready.
